src/PostgREST/App.hs

```diff
@@ -220,8 +219,23 @@ app dbStructure conf apiRequest =
     (_, _, _) -> return notFound
 
  where
+  toTableInfo :: [Table] -> [(Table, [Column], [Text])]
+  toTableInfo = map (\t ->
```

Comment:
While this function (along with encodeApi) tries to use available functionality (and it's the right thing to do at this point). the way it's implemented is quite inefficient (2 queries for each table).
In the future it should expect to receive a single "DbStructure" object based on which the schema is generated (for different output formats)